src/main/java/com/bandwidth/iris/sdk/IrisClient.java

```diff
@@ -1,186 +0,0 @@
-package com.bandwidth.iris.sdk;
-
-import com.bandwidth.iris.sdk.model.BaseModel;
-import com.bandwidth.iris.sdk.model.BaseResponse;
-import com.bandwidth.iris.sdk.utils.XmlUtils;
-import org.apache.commons.lang3.StringUtils;
-import org.apache.http.Header;
-import org.apache.http.HttpResponse;
-import org.apache.http.auth.AuthScope;
-import org.apache.http.auth.Credentials;
-import org.apache.http.auth.UsernamePasswordCredentials;
-import org.apache.http.client.methods.*;
-import org.apache.http.client.utils.URIBuilder;
-import org.apache.http.entity.ContentType;
-import org.apache.http.entity.FileEntity;
-import org.apache.http.entity.StringEntity;
-import org.apache.http.impl.client.DefaultHttpClient;
-import org.apache.http.util.EntityUtils;
-
-import javax.xml.stream.XMLInputFactory;
-import java.io.File;
-import java.net.URISyntaxException;
-import java.util.HashMap;
-import java.util.Map;
-
-/**
- * Created by sbarstow on 9/22/14.
- */
-public class IrisClient {
-
-    private static String defaultUri = "https://api.inetwork.com/v1.0";
-    private static String defaultVersion = "v1.0";
-    protected DefaultHttpClient httpClient;
-    private String baseUserUrl;
-    private String baseUrl;
-    private String accountId;
-    private String uri;
-    private String userName;
-    private String password;
-    private String version;
-    private XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
-
-    public IrisClient(String uri, String accountId,
-            String userName, String password, String version) {
-        this.uri = uri;
-        this.accountId = accountId;
-        this.userName = userName;
-        this.password = password;
-        this.version = version;
-
-        httpClient = new DefaultHttpClient();
-        Credentials credentials = new UsernamePasswordCredentials(userName, password);
-        httpClient.getCredentialsProvider().setCredentials(AuthScope.ANY, credentials);
-        this.baseUrl = "/" + version + "/";
-        this.baseUserUrl = this.baseUrl + "accounts/" + this.accountId + "/";
-
-    }
-
-    public IrisClient(String accountId, String userName, String password) {
-        this(defaultUri, accountId, userName, password, defaultVersion);
-    }
-
-    private <T> T processResponse(String responseBody, Class<T> returnType) throws Exception {
```

Comment:
Throwing plain `Exception` is considered impolite. It hides what the actual exceptions can be and where they can come from. Either catch all exceptions and re-throw them as RuntimeException or let the actual Exception trickle up but leave the exception type in the signature. It might be code-smell if a method can throw too many or too disparate exceptions. The method might just be doing too much.